File: src/main/java/fi/aalto/cs/apluscourses/intellij/notifications/CourseFileError.java
package fi.aalto.cs.apluscourses.intellij.notifications;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getAndReplaceText;
import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

public class CourseFileError extends Notification {

  /**
   * Error to be shown when course settings file cannot be accessed.
   *
   * @param e Exception.
   */
  public CourseFileError(Exception e) {
    super(PluginSettings.A_PLUS,
        getText("notification.CourseFileError.title"),
        getAndReplaceText("notification.CourseFileError.content", e.getMessage()),
        NotificationType.ERROR);
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/notifications/CourseFileError.kt
package fi.aalto.cs.apluscourses.notifications

import com.intellij.notification.Notification
import com.intellij.notification.NotificationType
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.services.PluginSettings

class CourseFileError
/**
 * Error to be shown when course settings file cannot be accessed.
 *
 * @param e Exception.
 */
    (e: Exception) : Notification(
    PluginSettings.A_PLUS,
    MyBundle.message("notification.CourseFileError.title"),
    MyBundle.message("notification.CourseFileError.content", e.message!!),
    NotificationType.ERROR
)
